lib/spack/spack/extensions.py

```diff
@@ -27,28 +65,26 @@ def extension_name(path):
         The extension name or None if path doesn't match the format
         for Spack's extension.
     """
-    regexp_match = re.search(extension_regexp, os.path.basename(path))
+    regexp_match = re.search(_extension_regexp, os.path.basename(path))
     if not regexp_match:
-        msg = "[FOLDER NAMING]"
-        msg += " {0} doesn't match the format for Spack's extensions"
-        tty.warn(msg.format(path))
-        return None
+        raise ExtensionNamingError(path)
```

Comment:
Is there a reason for the change in behavior? Also, the docstring is out of sync with the behavior of the function now.

Reply:
In general, I try to make, "I couldn't do the one thing you asked" an error rather than a warning where there isn't an obvious good thing to do instead. Given the multiple (including very early, during building of the  extension command cache) places this could happen, I thought this would be better as an exception, and it was more consistent with other error conditions like `CommandNotFoundError`, below.